File: binding/src/libqradiobutton.cpp
#include <QAbstractButton>
#include <QEvent>
#include <QMouseEvent>
#include <QPaintEvent>
#include <QPoint>
#include <QRadioButton>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyleOptionButton>
#include <QWidget>
#include <qradiobutton.h>
#include "libqradiobutton.h"
#include "libqradiobutton.hxx"

QRadioButton* QRadioButton_new(QWidget* parent) {
	 return new VirtualQRadioButton(parent);
}

QRadioButton* QRadioButton_new2() {
	 return new VirtualQRadioButton();
}

QRadioButton* QRadioButton_new3(const libqt_string text) {
	 return new VirtualQRadioButton(QString::fromUtf8(text.data, text.len));
}

QRadioButton* QRadioButton_new4(const libqt_string text, QWidget* parent) {
	 return new VirtualQRadioButton(QString::fromUtf8(text.data, text.len), parent);
}

libqt_string QRadioButton_Tr(const char* s) {
	QString _ret = QRadioButton::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QSize* QRadioButton_SizeHint(const QRadioButton* self) {
	return new QSize(self->sizeHint());
}

QSize* QRadioButton_MinimumSizeHint(const QRadioButton* self) {
	return new QSize(self->minimumSizeHint());
}

libqt_string QRadioButton_Tr2(const char* s, const char* c) {
	QString _ret = QRadioButton::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QRadioButton_Tr3(const char* s, const char* c, int n) {
	QString _ret = QRadioButton::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QSize* QRadioButton_QBaseSizeHint(const QRadioButton* self) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_SizeHint_IsBase(true);
	return new QSize(vqradiobutton->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnSizeHint(const QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_SizeHint_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_SizeHint_Callback>(slot));
}
}

// Base class handler implementation
QSize* QRadioButton_QBaseMinimumSizeHint(const QRadioButton* self) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_MinimumSizeHint_IsBase(true);
	return new QSize(vqradiobutton->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnMinimumSizeHint(const QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_MinimumSizeHint_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_MinimumSizeHint_Callback>(slot));
}
}

// Derived class handler implementation
bool QRadioButton_Event(QRadioButton* self, QEvent* e) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
	return vqradiobutton->event(e);
	} else {
	return self->QRadioButton::event(e);
}
}

// Base class handler implementation
bool QRadioButton_QBaseEvent(QRadioButton* self, QEvent* e) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_Event_IsBase(true);
	return vqradiobutton->event(e);
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnEvent(QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_Event_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_Event_Callback>(slot));
}
}

// Derived class handler implementation
bool QRadioButton_HitButton(const QRadioButton* self, const QPoint* param1) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
	return vqradiobutton->hitButton(*param1);
	} else {
	return self->QRadioButton::hitButton(*param1);
}
}

// Base class handler implementation
bool QRadioButton_QBaseHitButton(const QRadioButton* self, const QPoint* param1) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_HitButton_IsBase(true);
	return vqradiobutton->hitButton(*param1);
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnHitButton(const QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_HitButton_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_HitButton_Callback>(slot));
}
}

// Derived class handler implementation
void QRadioButton_PaintEvent(QRadioButton* self, QPaintEvent* param1) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
	vqradiobutton->paintEvent(param1);
	} else {
	self->QRadioButton::paintEvent(param1);
}
}

// Base class handler implementation
void QRadioButton_QBasePaintEvent(QRadioButton* self, QPaintEvent* param1) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_PaintEvent_IsBase(true);
	vqradiobutton->paintEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnPaintEvent(QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_PaintEvent_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_PaintEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QRadioButton_MouseMoveEvent(QRadioButton* self, QMouseEvent* param1) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
	vqradiobutton->mouseMoveEvent(param1);
	} else {
	self->QRadioButton::mouseMoveEvent(param1);
}
}

// Base class handler implementation
void QRadioButton_QBaseMouseMoveEvent(QRadioButton* self, QMouseEvent* param1) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_MouseMoveEvent_IsBase(true);
	vqradiobutton->mouseMoveEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnMouseMoveEvent(QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_MouseMoveEvent_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_MouseMoveEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QRadioButton_InitStyleOption(const QRadioButton* self, QStyleOptionButton* button) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
	vqradiobutton->initStyleOption(button);
	} else {
	self->QRadioButton::initStyleOption(button);
}
}

// Base class handler implementation
void QRadioButton_QBaseInitStyleOption(const QRadioButton* self, QStyleOptionButton* button) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_InitStyleOption_IsBase(true);
	vqradiobutton->initStyleOption(button);
}
}

// Auxiliary method to allow providing re-implementation
void QRadioButton_OnInitStyleOption(const QRadioButton* self, intptr_t slot) {
	auto* vqradiobutton = dynamic_cast<const VirtualQRadioButton*>(self);
	if (vqradiobutton && vqradiobutton->isVirtualQRadioButton) {
vqradiobutton->setQRadioButton_InitStyleOption_Callback(reinterpret_cast<VirtualQRadioButton::QRadioButton_InitStyleOption_Callback>(slot));
}
}

void QRadioButton_Delete(QRadioButton* self) {
    delete self;
}

File: binding/src/libqradiobutton.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qradiobutton.html
pub const qradiobutton = struct {

    /// New constructs a new QRadioButton object.
    pub fn New(parent: ?*anyopaque) QtC.QRadioButton {
        return qtc.QRadioButton_new(@ptrCast(parent));
    }


    /// New2 constructs a new QRadioButton object.
    pub fn New2() QtC.QRadioButton {
        return qtc.QRadioButton_new2();
    }


    /// New3 constructs a new QRadioButton object.
    pub fn New3(text: []const u8) QtC.QRadioButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QRadioButton_new3(text_str);
    }


    /// New4 constructs a new QRadioButton object.
    pub fn New4(text: []const u8, parent: ?*anyopaque) QtC.QRadioButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QRadioButton_new4(text_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRadioButton_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qradiobutton.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QRadioButton_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QRadioButton_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QRadioButton_QBaseSizeHint(@ptrCast(self));
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QRadioButton_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QRadioButton_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QRadioButton_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn Event(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QRadioButton_Event(@ptrCast(self), @ptrCast(e));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QRadioButton_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QRadioButton_QBaseEvent(@ptrCast(self), @ptrCast(e));
    }

    pub fn HitButton(self: ?*anyopaque, param1: ?*anyopaque) bool {
        return qtc.QRadioButton_HitButton(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnHitButton(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QRadioButton_OnHitButton(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseHitButton(self: ?*anyopaque, param1: ?*anyopaque) bool {
        return qtc.QRadioButton_QBaseHitButton(@ptrCast(self), @ptrCast(param1));
    }

    pub fn PaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QRadioButton_PaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnPaintEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRadioButton_OnPaintEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QRadioButton_QBasePaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn MouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QRadioButton_MouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnMouseMoveEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRadioButton_OnMouseMoveEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QRadioButton_QBaseMouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn InitStyleOption(self: ?*anyopaque, button: ?*anyopaque) void {
        qtc.QRadioButton_InitStyleOption(@ptrCast(self), @ptrCast(button));
    }

    /// Allows for overriding the related default method
    pub fn OnInitStyleOption(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRadioButton_OnInitStyleOption(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInitStyleOption(self: ?*anyopaque, button: ?*anyopaque) void {
        qtc.QRadioButton_QBaseInitStyleOption(@ptrCast(self), @ptrCast(button));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRadioButton_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qradiobutton.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRadioButton_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qradiobutton.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QRadioButton_Delete(@ptrCast(self));
    }
};
